rhp/v3/transport.go

```diff
@@ -48,43 +54,50 @@ type protocolObject interface {
 	encoding.SiaUnmarshaler
 }
 
-func writeRequest(s *mux.Stream, id Specifier, req protocolObject) error {
-	if _, err := s.Write(id[:]); err != nil {
+func writeRequest(w io.Writer, id Specifier, req protocolObject) error {
```

Comment:
Changed a bunch of signatures to `io.Reader` and `io.Writer` to allow for the `stream` wrapper.